File: training/ruby/lib/de/sots/cellars_and_centaurs/domain/model/trap.rb
# frozen_string_literal: true

module De::Sots::CellarsAndCentaurs::Domain::Model
  class Trap
    extend Fightable

    def self.attack(target)
      target.hit_points = target.hit_points.take_damage(3)
    end

    def self.initiative
      0
    end
  end
end

File: training/java/src/main/java/de/sots/cellarsandcentaurs/application/Trap.java
package de.sots.cellarsandcentaurs.application;

import de.sots.cellarsandcentaurs.domain.model.Fightable;

public class Trap {
    private static final int TRAP_INITIATIVE = 20;

    public Object arm() {
        return new Fightable() {
            @Override
            public int initiative() {
                return TRAP_INITIATIVE;
            }
        };
    }
}
